connectors/sources/google_drive.py

```diff
@@ -0,0 +1,638 @@
+            },
+            "retry_count": {
+                "default_value": DEFAULT_RETRY_COUNT,
+                "display": "numeric",
+                "label": "Maximum retries for failed requests",
+                "order": 2,
+                "required": False,
+                "type": "int",
+                "ui_restrictions": ["advanced"],
+                "value": DEFAULT_RETRY_COUNT,
+            },
+        }
+
+    @cached_property
+    def _google_drive_client(self):
+        """Initialize and return the GoogleDriveClient
+
+        Returns:
+            GoogleDriveClient: An instance of the GoogleDriveClient.
+        """
+        json_credentials = json.loads(self.configuration["service_account_credentials"])
+
+        if (
+            json_credentials.get("private_key")
+            and "\n" not in json_credentials["private_key"]
+        ):
+            json_credentials["private_key"] = get_pem_format(
+                key=json_credentials["private_key"].strip(),
+                max_split=2,
+            )
+
+        return GoogleDriveClient(
+            json_credentials=json_credentials,
+            retry_count=self.configuration["retry_count"],
+        )
+
+    async def ping(self):
+        """""Verify the connection with Google Drive""" ""
+        if RUNNING_FTEST:
+            return
+
+        try:
+            await anext(
+                self._google_drive_client.api_call(
+                    resource="about", method="get", fields="kind"
+                )
+            )
+            logger.info("Successfully connected to the Google Drive.")
+        except Exception:
+            logger.exception("Error while connecting to the Google Drive.")
+            raise
+
+    async def validate_config(self):
+        """Validates whether user inputs are valid or not for configuration field.
+
+        Raises:
+            Exception: The format of service account json is invalid.
+        """
+        self.configuration.check_valid()
+
+        try:
+            json.loads(self.configuration["service_account_credentials"])
+        except ValueError as e:
+            raise ConfigurableFieldValueError(
+                "Google Drive service account is not a valid JSON."
+            ) from e
+
+    async def get_drives(self):
+        """Fetch all shared drive (id, name) from Google Drive
+
+        Yields:
+            dict: Shared drive metadata.
+        """
+
+        async for drive in self._google_drive_client.api_call(
+            resource="drives",
+            method="list",
+            fields="nextPageToken,drives(id,name)",
+            full_response=True,
+            pageSize=100,
+        ):
+            yield drive
+
+    async def retrieve_all_drives(self):
+        """Retrieves all shared drives from Google Drive
+
+        Returns:
+            dict: mapping between drive id and its name
+        """
+        drives = {}
+        async for chunk in self.get_drives():
```

Comment:
```suggestion
        async for page in self.get_drives():
```

It's probably page, right?

Chunk is more about files IMO